qutebrowser/browser/browsertab.py

```diff
@@ -139,6 +137,10 @@ def run_string(self, name):
             raise WebTabError("{} is not a valid web action!".format(name))
         self._widget.triggerPageAction(member)
 
+    def show_source(self, win_id, url):
```

Comment:
You shouldn't pass the `win_id` and `url` here, as the tab knows both of those things itself, and this API should be as easy to use as possible (it's going to be used in a plugin API in the future).

Instead, pass the tab object when creating `AbstractAction` or its subclasses (see e.g. `AbstractCaret`) and then use `self._tab.win_id` and `self._tab.url()` or so.